src/core/OIDNDenoiser.js

```diff
@@ -0,0 +1,297 @@
+// Utils to interface with the denoiser
+
+import { Denoiser } from 'denoiser';
+import { FullScreenQuad } from 'three/examples/jsm/postprocessing/Pass.js';
+import { ClampedInterpolationMaterial } from '../materials/fullscreen/ClampedInterpolationMaterial.js';
+import { DenoiserBlendMaterial } from '../materials/fullscreen/DenoiserBlendMaterial.js';
+
+import { NoBlending, WebGLRenderTarget, SRGBColorSpace } from 'three';
+
+export class OIDNDenoiser {
+
+	get weightsUrl() {
+
+		return this.denoiser.weightsUrl;
+
+	}
+
+	set weightsUrl( url ) {
+
+		this.denoiser.weightsUrl = url;
+
+	}
+
+	get quality() {
+
+		return this.denoiser.quality;
+
+	}
+
+	set quality( v ) {
+
+		this.denoiser.quality = v;
+		if ( v === 'fast' && this.hdr ) this.hdr = false;
+		//if ( v === 'balanced' && ! this.hdr ) this.hdr = true;
+
+	}
+
+	get useAux() {
+
+		return this._useAux;
+
+	}
+
+	set useAux( v ) {
+
+		this._useAux = v;
+		this.denoiser.resetInputs();
+
+	}
+
+	get cleanAux() {
+
+		return ! this.denoiser.dirtyAux;
+
+	}
+
+	set cleanAux( v ) {
+
+		this.denoiser.dirtyAux = ! v;
+
+	}
+
+	get hdr() {
+
+		return this.denoiser.hdr;
+
+	}
```

Comment:
Is `hdr` functional right now? It looks like we always pass sRGB into the denoiser

Reply:
It was.
Technically the denosier can already support it.
While working out the color issues I matching the original system as much as possible. this included passing the hdr texture. 

When passing HDR and using "fast" quality I got HORRIBLE results, but using balanced was excellent. That's why I added the auto change to those props.

When trying to address the color issues in LDR I resolved it and found the system to be more stable with fast/balanced quality while using LDR.

If you don't ever plan to pass HDR to the denoiser or don't like leaving future handling code I can remove it.

Reply:
Gotcha - is it the case that we just have to pass the original LinearSRGB Path Traced texture when `hdr` is true and we'll get a LinearSRGB texture out? I can try it a bit to see how it looks.

Reply:
Yes, if we pass that texture as the input the denoiser will return it in the same range (or at least it should, I've only tested a few times)
OIDN says that in HDR 1 should be a luminance level of 100 cd/m^2
so for best results it should be similar.

If you do that you may want to disable the SRGBToLinear conversion that takes place on the denoiser result. I do this so we can correctly apply the ToneMapper.

One thing to consider would be dropping the SRGB > Denoiser requirement. This would mean textures dont need to be converted when returned and can be tonemapped directly